src/coreComponents/constitutive/solid/DruckerPragerExtended.hpp

```diff
@@ -0,0 +1,495 @@
+
+  /// Default move constructor
+  DruckerPragerExtendedUpdates( DruckerPragerExtendedUpdates && ) = default;
+
+  /// Deleted default constructor
+  DruckerPragerExtendedUpdates() = delete;
+
+  /// Deleted copy assignment operator
+  DruckerPragerExtendedUpdates & operator=( DruckerPragerExtendedUpdates const & ) = delete;
+
+  /// Deleted move assignment operator
+  DruckerPragerExtendedUpdates & operator=( DruckerPragerExtendedUpdates && ) =  delete;
+
+  /// Use the uncompressed version of the stiffness bilinear form
+  using DiscretizationOps = SolidModelDiscretizationOpsFullyAnisotroipic; // TODO: typo in anistropic (fix in DiscOps PR)
+
+  // Bring in base implementations to prevent hiding warnings
+  using ElasticIsotropicUpdates::smallStrainUpdate;
+
+  GEOSX_HOST_DEVICE
+  virtual void smallStrainUpdate( localIndex const k,
+                                  localIndex const q,
+                                  real64 const ( &strainIncrement )[6],
+                                  real64 ( &stress )[6],
+                                  real64 ( &stiffness )[6][6] ) const override final;
+
+  GEOSX_HOST_DEVICE
+  virtual void smallStrainUpdate( localIndex const k,
+                                  localIndex const q,
+                                  real64 const ( &strainIncrement )[6],
+                                  real64 ( &stress )[6],
+                                  DiscretizationOps & stiffness ) const final;
+
+private:
+  /// A reference to the ArrayView holding the initial friction angle for each element.
+  arrayView1d< real64 const > const m_initialFriction;
+
+  /// A reference to the ArrayView holding the residual friction angle for each element.
+  arrayView1d< real64 const > const m_residualFriction;
+
+  /// A reference to the ArrayView holding the dilation ratio for each element.
+  arrayView1d< real64 const > const m_dilationRatio;
+
+  /// A reference to the ArrayView holding the pressure intercept for each element.
+  arrayView1d< real64 const > const m_pressureIntercept;
+
+  /// A reference to the ArrayView holding the hardening parameter for each element.
+  arrayView1d< real64 const > const m_hardening;
+
+  /// A reference to the ArrayView holding the new state variable for each integration point
+  arrayView2d< real64 > const m_newState;
+
+  /// A reference to the ArrayView holding the old state variable for each integration point
+  arrayView2d< real64 > const m_oldState;
+
+  /// Hyperbolic model for friction hardening
+  GEOSX_HOST_DEVICE
+  void hyperbolicModel( real64 const y1,
+                        real64 const y2,
+                        real64 const m,
+                        real64 const x,
+                        real64 & y,
+                        real64 & dy_dx ) const
+  {
+    if( x<0 )
+    {
+      y = y1;
+      dy_dx = 0;
+    }
+    else
+    {
+      y = y1 + (y2-y1)*x/(m+x);
+      dy_dx = (y2-y1)*m/((m+x)*(m+x));
+    }
```

Comment:
We have a division to zero issue here if x=0 and users set zero hardening condition (m=0). I suggest to use a tolerance parameter here for the the condition, e.g. ( x<1e-10 ) in place of ( x<0 )